src/scenes/Event/store.js

```diff
@@ -295,15 +306,28 @@ export default class EventStore {
 
   @action
   queryTransactions = async (address) => {
-    this.transactions = await queryAllTransactions(
-      [{ topicAddress: address }],
+    const pendings = await queryAllTransactions(
+      [{ topicAddress: address, status: TransactionStatus.PENDING }],
       { field: 'createdTime', direction: SortBy.DESCENDING },
     );
+    this.withdraws = await queryWithdraws([{ topicAddress: address }]);
+
+    this.finalizes = await queryFinalizes([{ topicAddress: address }]);
+    this.finalizes = filter(this.finalizes, { oracleAddress: null });
```

Comment:
do you need to pass in null as `oracleAddress` or can you just leave out the filter entirely?

Reply:
`oracleAddress == null` is the rule to say if it's `finalize`, right? The result from the query would give both `resultset` and `finalize` belong to that `topicAddress`, I only want to get the `finalize` tho.

Reply:
Yes you are right. Was wondering why the finalize doesn't have an oracleAddress so I checked the contract code and its cause it is executed in the TopicEvent. Bad design on my part when I created the contracts, should've had separate events for ResultSet and Finalize.